providers/base.ts

```diff
@@ -351,6 +351,48 @@ export abstract class ReleaseLookup<Provider extends MetadataProvider, RawReleas
 	}
 }
 
+/** Extends MetadataProvider with functions common to lookups accessing web APIs. */
+export abstract class MetadataApiProvider extends MetadataProvider {
+	/** Must be implemented to perform a request against the specified URL. */
+	abstract query<Data>(apiUrl: URL, maxTimestamp?: number): Promise<CacheEntry<Data>>;
+}
+
+/** Extends ReleaseLookup with functions common to lookups accessing web APIs. */
+export abstract class ReleaseApiLookup<Provider extends MetadataApiProvider, RawRelease>
+	extends ReleaseLookup<Provider, RawRelease> {
+	/** Constructs an API URL for a release using the specified lookup options. */
+	abstract constructReleaseApiUrl(): URL;
+
+	/** Performs the query for the URL returned by constructReleaseApiUrl for all configured regions. */
```

Comment:
```suggestion
	/** Performs the query for the URL returned by {@linkcode constructReleaseApiUrl} for all configured regions until valid data is returned. */
```
Untested, but this should make IDE tooltips link to the method (in VS Code).

Reply:
Indeed it does!